### PR TITLE
`General/Bugfix` fix null-Feedbacks

### PR DESCRIPTION
**Pull request by Shirkanesi**
As described in #151 weird things go on where the feedbacks will contain null-values which introduce some weird bugs in the plugin.
I'm currently trying to find all the issues that (might) break by null-Feedbacks, however it's not realy clear where exactly everything breaks. 
To increase chances of success, I also added proper logging to `ViewAlertObserver` in order to get more information about the real error happening somewhere.

I'm realy interested in your oppinions whether we should directly add this logging to production or we should first investigate more before.

Also I'd realy like to hear your oppinions about where even more null-checks might be necessary.

### CODE REVIEW

**Comment by Shirkanesi** [comment]
Draft for now, however your oppinion is required @dfuchss

**Comment by dfuchss** [comment]
looks good to me.

**Comment by Shirkanesi** [comment]
I tested around even more. I've managed to reproduce the bug at a relatively consistant rate.
It seems like the issue with `A is null` happens everytime an eclipse-instance uploads new assessments. However sometimes those uploads will fail with `400 Bad Request`. In this case the data seems to be not updated yet but parts of the feedbacks in the database are broken.

This 400 seems to be triggered by a (too) big request of some kind. However I'm not exactly sure what exactly breaks here BUT it is pretty sure an issue in the grading-plugin.

A temporary fix for this issue is clearing the current feedback and rewriting everything. This clearing can be done by an Instructor OR by aborting the assessment (if the button is still working...).

**Comment by Shirkanesi** [comment]
Finally found the issue; `createNewManualUnreferencedFeedback` created to large Feedbacks...
Now fixed by creating multiple annotations.

File: bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/observers/ViewAlertObserver.java [hunk 1]
```diff
@@ -11,19 +13,24 @@
  *
  */
 public class ViewAlertObserver implements IAlertObserver {
-
+	
+	private static final ILog log = Platform.getLog(ViewAlertObserver.class);
```
Comment (dfuchss) [inline bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/observers/ViewAlertObserver.java:17]: Logging is good!

File: bundles/edu.kit.kastel.sdq.eclipse.grading.core/src/edu/kit/kastel/sdq/eclipse/grading/core/artemis/AnnotationMapper.java [hunk 2]
```diff
@@ -234,24 +237,50 @@ private Feedback createNewManualUnreferencedFeedback(IRatingGroup ratingGroup) {
...
 		if (this.penaltyCalculationStrategy.penaltyLimitIsHitForRatingGroup(ratingGroup)) {
-			detailText += "\n    * Note: The sum of penalties hit the penalty limit for this rating group.";
+			lines.add("\n    * Note: The sum of penalties hit the penalty limit for this rating group.");
 		}
 
-		return new Feedback(FeedbackType.MANUAL_UNREFERENCED.name(), calculatedPenalty, null, null, null, null, null, detailText);
+		
+		List<String> feedbackTexts = new LinkedList<>();
+		
+		if (lines.size() == 0) {
```
Comment (dfuchss) [inline bundles/edu.kit.kastel.sdq.eclipse.grading.core/src/edu/kit/kastel/sdq/eclipse/grading/core/artemis/AnnotationMapper.java:260]: isEmpty :) ?
```suggestion
		if (lines.isEmpty()) {

```

File: bundles/edu.kit.kastel.sdq.eclipse.grading.core/src/edu/kit/kastel/sdq/eclipse/grading/core/artemis/AnnotationMapper.java [hunk 3]
```diff
@@ -234,24 +237,50 @@ private Feedback createNewManualUnreferencedFeedback(IRatingGroup ratingGroup) {
...
+			return List.of();
+		}
+		
+		String text = annotationHeadline + " (annotation " + 1 + ")";
+		
+		for (int i = 0; i < lines.size(); i++) {
+			if (text.length() + lines.get(i).length() >= FEEDBACK_DETAIL_TEXT_MAX_CHARACTERS - annotationHeadline.length() - 50) {
```
Comment (dfuchss) [inline bundles/edu.kit.kastel.sdq.eclipse.grading.core/src/edu/kit/kastel/sdq/eclipse/grading/core/artemis/AnnotationMapper.java:267]: Why `50`?

**Review by dfuchss — CHANGES_REQUESTED** [review]
Looks good to me. See remarks :)

File: bundles/edu.kit.kastel.sdq.eclipse.grading.core/src/edu/kit/kastel/sdq/eclipse/grading/core/artemis/AnnotationMapper.java (hunk 3 above)
Comment (dfuchss) [reply bundles/edu.kit.kastel.sdq.eclipse.grading.core/src/edu/kit/kastel/sdq/eclipse/grading/core/artemis/AnnotationMapper.java:267]: Yes you are right :) .. Maybe add it to a constant as well :)

**Comment by dfuchss** [comment]
Works as described. Thanks :)

**Pull request merged**